src/core_plugins/kibana/public/dashboard/top_nav/save_modal.js

```diff
@@ -0,0 +1,252 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { Fragment } from 'react';
+import PropTypes from 'prop-types';
+
+import {
+  EuiButton,
+  EuiFieldText,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiOverlayMask,
+  EuiSpacer,
+  EuiCallOut,
+  EuiForm,
+  EuiFormRow,
+  EuiTextArea,
+  EuiSwitch,
+} from '@elastic/eui';
+
+export class DashboardSaveModal extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      title: props.title,
+      description: props.description,
+      copyOnSave: props.copyOnSave,
+      timeRestore: props.timeRestore,
+      isTitleDuplicateConfirmed: false,
+      hasTitleDuplicate: false,
+      isLoading: false,
+    };
+  }
+  componentDidMount() {
+    this._isMounted = true;
+  }
+
+  componentWillUnmount() {
+    this._isMounted = false;
+  }
+
+  onTitleDuplicate = () => {
+    this.setState({
+      isTitleDuplicateConfirmed: true,
+      hasTitleDuplicate: true,
+    });
+  }
+
+  saveDashboard = async () => {
+    this.setState({
+      isLoading: true,
+    });
+
+    await this.props.onSave(
+      this.state.title,
+      this.state.description,
+      this.state.copyOnSave,
+      this.state.timeRestore,
+      this.state.isTitleDuplicateConfirmed,
+      this.onTitleDuplicate);
+
+    if (this._isMounted) {
+      this.setState({
+        isLoading: false,
+      });
+    }
+  };
+
+  onTitleChange = (event) => {
+    this.setState({
+      title: event.target.value,
+      isTitleDuplicateConfirmed: false,
+      hasTitleDuplicate: false,
+    });
+  };
+
+  onDescriptionChange = (event) => {
+    this.setState({
+      description: event.target.value,
+    });
+  };
+
+  onCopyOnSaveChange = (event) => {
+    this.setState({
+      copyOnSave: event.target.checked,
+    });
+  }
+
+  onTimeRestoreChange = (event) => {
+    this.setState({
+      timeRestore: event.target.checked,
+    });
+  }
+
+  renderDuplicateTitleCallout = () => {
+    if (!this.state.hasTitleDuplicate) {
+      return;
+    }
+
+    return (
+      <Fragment>
+        <EuiCallOut
+          title={`A Dashboard with the title '${this.state.title}' already exists.`}
+          color="warning"
+          data-test-subj="titleDupicateWarnMsg"
+        >
+          <p>
+            Click <strong>Confirm Save</strong> to save the dashboard with the duplicate title.
+          </p>
+        </EuiCallOut>
+        <EuiSpacer />
+      </Fragment>
+    );
+  }
+
+  renderCopyOnSave = () => {
+    if (!this.props.showCopyOnSave) {
+      return;
+    }
+
+    let helpText;
+    if (this.props.title !== this.state.title) {
+      helpText = `In previous versions of Kibana, changing the name of a dashboard would make a copy with the new name.
```

Comment:
I could have sworn there is an issue somewhere that says to get rid of this after 6.0 or something because the warning was only needed in 5.x but I can't find it. I think this warning has been here for long enough that we don't need to call out the "previous functionality" - it was in pre 6.0, and since major's can have BWC breaking changes...